front/lib/actions/mcp_internal_actions/index.ts

```diff
@@ -22,7 +23,13 @@ export const isEnabledForWorkspace = async (
   if (mcpServer.isRestricted) {
     const featureFlags = await getFeatureFlags(auth.getNonNullableWorkspace());
     const plan = auth.getNonNullablePlan();
-    return !mcpServer.isRestricted({ plan, featureFlags });
+    const isDustDeepDisabled = await isDustDeepDisabledByAdmin(auth);
```

Comment:
The only annoying thing it it makes a query on db each time. Would you have a better idea @fontanierh? 🙏🏻